### PR TITLE
Fix Studio auth hook parsing for http URIs

### LINKED ISSUE #46950: Studio misclassifies http:// auth hooks as Postgres hooks
Summary

Studio treats any non-https auth hook URI as a Postgres hook. That breaks self-hosted HTTP hooks, including the Send Email example shipped in docker/docker-compose.yml.

Evidence

- hooks.utils.ts only special-cases URIs that start with "https" and sends every other URI through the Postgres parser: https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/apps/studio/components/interfaces/Auth/Hooks/hooks.utils.ts#L11-L19
- With the documented self-hosted example "http://host.docker.internal:54321/functions/v1/email_sender", that parser produces a Postgres hook shape with schema "functions" and function name "v1".
- CreateHookSheet.tsx only accepts "https://" in the HTTP branch, so Studio cannot preserve or edit an existing "http://" hook: https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx#L81-L87
- The same sheet later rewrites any Postgres selection back to a pg-functions URI: https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx#L238-L248
- HooksListing.tsx runs revoke SQL for hooks classified as Postgres, so deleting the misparsed HTTP hook targets function "functions.v1": https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/apps/studio/components/interfaces/Auth/Hooks/HooksListing.tsx#L52-L59 and https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/apps/studio/components/interfaces/Auth/Hooks/HooksListing.tsx#L202-L217
- The shipped self-hosted compose example uses an HTTP Send Email hook URI: https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/docker/docker-compose.yml#L248-L250
- The docs also describe HTTP hooks as a supported auth hook mode: https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/apps/docs/content/guides/auth/auth-hooks.mdx#L316-L335 and https://github.com/supabase/supabase/blob/02e50aa86e6530891c2fc5e595ec8edd8aaf9201/apps/docs/content/guides/auth/auth-hooks/send-email-hook.mdx#L531-L565

Reproduction

1. Self-host Supabase with GOTRUE_HOOK_SEND_EMAIL_URI set to "http://host.docker.internal:54321/functions/v1/email_sender".
2. Open Authentication -> Hooks in Studio.
3. Open the Send Email hook.
4. Studio shows it as a Postgres hook instead of an HTTP hook.
5. The parsed values come out as schema "functions" and function name "v1".
6. If you save from Studio, it rewrites the URI to "pg-functions://postgres/functions/v1".
7. If you delete from Studio, it prepares revoke SQL against "functions.v1".

Expected behavior

Studio should treat both http:// and https:// hook URIs as HTTP hooks, preserve the full URL, and never generate Postgres grant or revoke SQL for them.

Impact

This breaks self-hosted HTTP auth hooks in the Studio UI and makes edit and delete flows unsafe for configurations that follow the shipped self-hosted example.

Suggested fix

Treat both "http://" and "https://" as HTTP hook URIs in extractMethod, allow the self-hosted HTTP case in the hook form validation, and add a regression test that covers the documented Send Email hook example.

### PR DESCRIPTION
**Pull request by peter941221**
Closes #46950

Studio currently treats any auth hook URI that does not start with `https` as a Postgres hook. That breaks the documented self-hosted Send Email example and can rewrite or revoke the wrong target.

This patch treats both `http://` and `https://` as HTTP hooks, keeps those URLs intact in the hook editor, renames the editor option from `HTTPS` to `HTTP`, and updates the custom access token warning to match the broader hook type.

The Postgres path stays unchanged. Postgres URIs still use the existing parser and still generate the same grant and revoke SQL.

Testing

From `apps/studio`:
`npm exec --yes pnpm@10.24.0 exec vitest run components/interfaces/Auth/Hooks/hooks.utils.test.ts --reporter verbose`
`npm exec --yes pnpm@10.24.0 exec prettier --check ./components/interfaces/Auth/Hooks/CreateHookSheet.tsx ./components/interfaces/Auth/Hooks/hooks.constants.ts ./components/interfaces/Auth/Hooks/hooks.utils.ts ./components/interfaces/Auth/Hooks/hooks.utils.test.ts ./components/interfaces/RoleImpersonationSelector/UserImpersonationSelector.tsx`
`./node_modules/.bin/tsc --noEmit -p tsconfig.json`

## Summary by CodeRabbit

* **Refactor**
  * Updated URL-based authentication hooks to use **HTTP** (instead of HTTPS) consistently across validation, form inputs, and submitted payloads, including URL detection for both `http://` and `https://` and correct secret handling/generation.

* **Bug Fixes**
  * Adjusted the impersonation info message to show correctly for **HTTP** custom access token hooks.

* **Tests**
  * Expanded unit tests to cover HTTP/HTTPS URL classification, validation behavior, and edge cases for hook extraction.

### CODE REVIEW

**Review by daltino — COMMENTED** [review]
Great fix! This correctly addresses a real user-facing bug where the documented self-hosted Send Email example (using `http://host.docker.internal:...`) was silently misclassified as a Postgres hook. The approach is clean, and the new `isHttpHookUrl` helper is a good abstraction. A few things are worth tightening before merge:

---

### 🐛 Potential Issue — Incomplete patch visibility

The diffs for `hooks.utils.ts` and `hooks.utils.test.ts` are truncated (the `@@ -2,14` and `@@ -0,0 +1,45` hunks cut off mid-sentence). I'll flag what I can see, but please make sure the full diff is included in the PR so reviewers can verify the implementation end-to-end.

---

### 🔴 Possible blocker — Legacy `'https'` type values may still exist elsewhere

The type literal changed from `'https'` to `'http'` in the interface and utility layer, but a codebase-wide search for `=== 'https'` or `type: 'https'` may reveal additional consumers that weren't updated.

For example:

* `CreateHookSheet.tsx` now imports `isHttpHookUrl`, but does it still contain any `type === 'https'` comparisons internally? (Not visible in the truncated diff.)
* Are there any API response parsers, serializers, localStorage entries, Zustand stores, or other persisted state that store/read the string `'https'`?

If hook configuration state is persisted anywhere, existing users could end up with stale `'https'` values that no longer match any branch after this change.

**Suggestion:** Run a repo-wide search for `'https'` in the Auth/Hooks area and any persistence layers before merging.

---

### 🟡 Minor — Confirm `extractMethod` uses `isHttpHookUrl`

The diff shows `isHttpHookUrl` being exported and imported in `CreateHookSheet.tsx`, but the `hooks.utils.ts` hunk is cut off before the body of `extractMethod` is visible.

Please confirm the original check was actually replaced:

```ts
export const extractMethod = (uri: string, secret?: string) => {
  if (isHttpHookUrl(uri)) {
    return { type: 'http', url: uri, secret: secret || '' }
  }

  // postgres fallback...
}
```

It would be easy to introduce the helper but accidentally leave the original `uri.startsWith('https')` logic in place, which would preserve the bug.

---

### 🟡 Minor — Test diff is truncated

The test file diff cuts off at line 18 (`"accepts both http and..."`). From what's visible, the structure looks solid—asserting that the documented `http://host.docker.internal` URL round-trips correctly is exactly the right regression test.

Please verify the final test suite covers:

1. ✅ `http://` URL → `{ type: 'http', ... }`
2. ✅ `https://` URL → `{ type: 'http', ... }`
3. ✅ Postgres function URI → `{ type: 'postgres', ... }`
4. ✅ Direct tests for `isHttpHookUrl()` if it is exported
5. ✅ Empty string and/or invalid URI inputs to verify fallback behavior and ensure no exceptions are thrown

---

### 🟢 Positive notes

* **`isHttpHookUrl`** is a clean, reusable abstraction and avoids duplicating URI detection logic.
* **Renaming the UI label** from `HTTPS` to `HTTP` is the correct UX change and aligns with the broader type rename.
* **`UserImpersonationSelector`** was updated in sync with the type change—easy detail to miss.
* The **`hooks.constants.ts`** interface update is minimal and focused.
* Adding a dedicated **`hooks.utils.test.ts`** file is a strong improvement; this utility clearly benefits from explicit coverage.

---

### Summary

The core logic fix is correct and addresses the reported bug. Before approving, I'd like to confirm:

1. No remaining `type === 'https'` checks or `'https'` type literals exist elsewhere in the codebase.
2. `extractMethod()` actually uses `isHttpHookUrl()` internally and no longer relies on `startsWith('https')`.
3. The test suite covers both HTTP and HTTPS URLs, Postgres URIs, and basic edge cases.
4. There are no compatibility concerns with previously persisted `'https'` values, if such state exists.

Assuming those checks pass, this looks ready to merge.

**Comment by peter941221** [comment]
Pushed 2764f2677f.

I checked the four points you called out.

- `extractMethod()` already routes both protocols through `isHttpHookUrl()` in `apps/studio/components/interfaces/Auth/Hooks/hooks.utils.ts:5-24`.
- `CreateHookSheet` and `useCustomAccessTokenHookDetails` both derive hook state from `extractMethod(...)`, so the editor and the custom access token hook path read the same branch (`apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx:75-149`, `apps/studio/hooks/misc/useCustomAccessTokenHookDetails.tsx:6-20`).
- I ran a targeted search in the hook-related Studio paths for `type: 'https'`, `=== 'https'`, and `selectedType.*https`. That returned no matches. I also did not find any hook-specific localStorage or persisted state for this type. The stored value is still the URI from auth config; the `'http'` / `'postgres'` branch is derived when Studio reads that URI.
- I pushed `test(studio): cover auth hook uri edge cases` to tighten `apps/studio/components/interfaces/Auth/Hooks/hooks.utils.test.ts:19-48`. The test file now covers `https://` URLs, Postgres URIs staying on the Postgres branch, and blank URI fallback without throwing, alongside the existing documented `http://host.docker.internal:54321/functions/v1/email_sender` regression.

Verification:
`cd apps/studio && npm exec --yes pnpm@10.24.0 exec vitest run components/interfaces/Auth/Hooks/hooks.utils.test.ts --reporter verbose`
`cd apps/studio && ../../node_modules/.bin/prettier --check ./components/interfaces/Auth/Hooks/hooks.utils.test.ts`

**Comment by Hallidayo** [comment]
Hi there, we move pretty fast here at Supabase, and we want to make sure that PRs that are relevant to the work get reviewed and worked on.

This PR is now over 60 days old, and from our understanding, it isn't addressing a bug or other active issue, so we're going to close it for now.

We do appreciate your time and contribution to Supabase :green_heart:
